### ISSUE TITLE
Invalid redefinition of TypedDict field not reported

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6810]
When multiple inheritance is used to define a TypedDict, conflicting field types should be reported.

Here is the example from the original PEP:

```python
class X2(TypedDict):
   x: int

class Y2(TypedDict):
   x: str

class XYZ2(X, Y):  # Type check error: cannot overwrite TypedDict field "x" while merging
   xyz: bool
```

**Comment by erictraut** [comment]
User error — pyright is doing the right thing here.
